.github/workflows/cask.yml

```diff
@@ -0,0 +1,35 @@
+name: cask
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 2
+    - name: Install dependencies
+      run: |
+        sudo apt-get install emacs25
+        sudo apt-get install git jq mercurial texinfo
+
+        # Work around SSL-related failures, see
+        # https://stackoverflow.com/questions/21477683/mercurial-https-clone-abort-error-wrong-version-number
+        echo "[ui]\ntls = False" > $HOME/.hgrc
+
+        # Use cask to install development dependencies
+        curl -fsSkL https://raw.github.com/cask/cask/master/go | python
+    - name: Run cask
+      run: |
+        ~/.cask/bin/cask
```

Comment:
Yeah, this could all be switched to use my `setup-emacs` action and Nix for `jq`, `mercurial` etc. in due course. But it's fine (and wise) to do what you're doing here as the first step.